core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -64,13 +71,31 @@ public class HTTPClient implements RESTClient {
   private final String uri;
   private final CloseableHttpClient httpClient;
   private final ObjectMapper mapper;
-  private final Map<String, String> baseHeaders;
```

Comment:
Default headers are now built as part of the client builder.  This should have no affect on behavior since they are immutable in this implementation.